Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp

```diff
@@ -479,13 +485,123 @@ namespace AzToolsFramework
+        {
+            return AZ::Failure(
+                AZStd::string::format(
+                    "Action Manager - Could not register mode \"%s\" for action context \"%s\" - mode with the same identifier already exists.",
+                    modeIdentifier.c_str(),
+                    actionContextIdentifier.c_str()
+                )
+            );
+        }
+
+        actionContextIterator->second->AddMode(modeIdentifier);
+        return AZ::Success();
+    }
+
+    ActionManagerOperationResult ActionManager::SetModeToAction(
+        const AZStd::string& modeIdentifier, const AZStd::string& actionIdentifier)
+    {
+        auto actionIterator = m_actions.find(actionIdentifier);
+        if (actionIterator == m_actions.end())
+        {
+            return AZ::Failure(
+                AZStd::string::format(
+                    "Action Manager - Could not set mode \"%s\" to action \"%s\" as no action with that identifier was registered.",
+                    modeIdentifier.c_str(),
+                    actionIdentifier.c_str()
+                )
+            );
+        }
+
+        auto actionContextIterator = m_actionContexts.find(actionIterator->second.GetActionContextIdentifier());
+        if (!actionContextIterator->second->HasMode(modeIdentifier))
+        {
+            return AZ::Failure(
+                AZStd::string::format(
+                    "Action Manager - Could not set mode \"%s\" to action \"%s\" as no mode with that identifier was registered.",
+                    modeIdentifier.c_str(),
+                    actionIdentifier.c_str()
+                )
+            );
+        }
+
+        actionIterator->second.SetMode(modeIdentifier);
+        return AZ::Success();
+    }
+
+    ActionManagerOperationResult ActionManager::SetActiveActionContextMode(
+        const AZStd::string& actionContextIdentifier, const AZStd::string& modeIdentifier)
+    {
+        auto actionContextIterator = m_actionContexts.find(actionContextIdentifier);
+        if (actionContextIterator == m_actionContexts.end())
+        {
+            return AZ::Failure(
+                AZStd::string::format(
+                    "Action Manager - Could not set active mode for action context \"%s\" to \"%s\" as the action context has not been registered.",
+                    actionContextIdentifier.c_str(),
+                    modeIdentifier.c_str()
+                )
+            );
+        }
+
+        if (!actionContextIterator->second->HasMode(modeIdentifier))
+        {
+            return AZ::Failure(
+                AZStd::string::format(
+                    "Action Manager - Could not set active mode for action context \"%s\" to \"%s\" as the mode has not been registered.",
+                    actionContextIdentifier.c_str(),
+                    modeIdentifier.c_str()
+                )
+            );
+        }
+
+        bool changed = actionContextIterator->second->SetActiveMode(modeIdentifier);
+        if (changed)
```

Comment:
uber nit: Could omit `changed` entirely and just use `if (actionContextIterator->second->SetActiveMode(modeIdentifier)`